src/pretix/control/views/organizer_views/organizer_view.py

```diff
@@ -317,3 +330,116 @@ def get_context_data(self, **kwargs):
     @cached_property
     def filter_form(self):
         return OrganizerFilterForm(data=self.request.GET, request=self.request)
+
+
+class BillingSettings(FormView, OrganizerPermissionRequiredMixin):
+    model = OrganizerBillingModel
+    form_class = BillingSettingsForm
+    template_name = "pretixcontrol/organizers/billing.html"
+    permission = "can_change_organizer_settings"
+
+    def get_success_url(self):
+        return reverse(
+            "control:organizer.settings.billing",
+            kwargs={
+                "organizer": self.request.organizer.slug,
+            },
+        )
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["organizer"] = self.request.organizer
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        ctx = super().get_context_data(**kwargs)
+
+        billing_settings = OrganizerBillingModel.objects.filter(
+            organizer_id=self.request.organizer.id
+        ).first()
+
+        if billing_settings and billing_settings.stripe_customer_id:
+            ctx["is_general_information_fulfilled"] = True
+        else:
+            ctx["is_general_information_fulfilled"] = False
+        return ctx
+
+    @staticmethod
+    def get_country_name(country_code):
+        country = CachedCountries().countries
+        return country.get(country_code, None)
+
+    def validate_vat_number(self, country_code, vat_number):
+        if country_code not in pyvat.VAT_REGISTRIES:
+            country_name = self.get_country_name(country_code)
+            messages.warning(self.request, _("VAT validation not supported for country: %s" % str(country_name)))
+            return True
+        result = pyvat.is_vat_number_format_valid(vat_number, country_code)
+        return result
+
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
+        form = self.get_form()
+
+        if form.is_valid():
+            cleaned_data = form.cleaned_data
+            country_code = cleaned_data.get("country")
+            vat_number = cleaned_data.get("tax_id")
+
+            if vat_number:
+                country_name = self.get_country_name(country_code)
+                is_valid_vat_number = self.validate_vat_number(country_code, vat_number)
+                if not is_valid_vat_number:
+                    messages.error(self.request, _("Invalid VAT number for country: %s" % str(country_name)))
+                    return self.form_invalid(form)
+
+            try:
+                form.save()
+                messages.success(self.request, _("Your changes have been saved."))
+                return redirect(self.get_success_url())
+            except ValidationError as e:
```

Comment:
You are using `transaction.atomic` in wrong way: You handle exception inside atomic block ([doc](https://docs.djangoproject.com/en/5.1/topics/db/transactions/#controlling-transactions-explicitly)).

Another issue, after calling `form.is_valid()`, you still doing more validation (`if vat_number`, `if not is_valid_vat_number`). It shows that you are using Django Form wrong way. Validations should be done in `form.is_valid()` call, not after it.

Reply:
Thank you for reviewing. I have resolved the issue.  
- I removed the use of transaction.atomic because I am only saving billing setting information into the billing settings table, which is independent of other tables. If an error occurs while adding data, it won't affect the other tables.
- I moved the tax_id validation into the is_valid function in the form.
Please check again. If you have any further questions, please let me know.